hs_enum/hs_enum.go

```diff
@@ -0,0 +1,598 @@
+	SHOWN_HERO_POWER GameTag = HERO_POWER
+	EVILZUG          GameTag = MARK_OF_EVIL
+
+	// Deleted
+	IGNORE_DAMAGE                      GameTag = 1
+	COPY_DEATHRATTLE                   GameTag = 55
+	COPY_DEATHRATTLE_INDEX             GameTag = 56
+	CARD_ID                            GameTag = 186
+	INCOMING_HEALING_MULTIPLIER        GameTag = 233
+	INCOMING_HEALING_ADJUSTMENT        GameTag = 234
+	INCOMING_HEALING_CAP               GameTag = 235
+	INCOMING_DAMAGE_MULTIPLIER         GameTag = 236
+	INCOMING_DAMAGE_ADJUSTMENT         GameTag = 237
+	INCOMING_DAMAGE_CAP                GameTag = 238
+	OUTGOING_DAMAGE_CAP                GameTag = 273
+	OUTGOING_DAMAGE_ADJUSTMENT         GameTag = 274
+	OUTGOING_DAMAGE_MULTIPLIER         GameTag = 275
+	OUTGOING_HEALING_CAP               GameTag = 276
+	OUTGOING_HEALING_ADJUSTMENT        GameTag = 277
+	OUTGOING_HEALING_MULTIPLIER        GameTag = 278
+	INCOMING_ABILITY_DAMAGE_ADJUSTMENT GameTag = 279
+	INCOMING_COMBAT_DAMAGE_ADJUSTMENT  GameTag = 280
+	OUTGOING_ABILITY_DAMAGE_ADJUSTMENT GameTag = 281
+	OUTGOING_COMBAT_DAMAGE_ADJUSTMENT  GameTag = 282
+	OUTGOING_ABILITY_DAMAGE_MULTIPLIER GameTag = 283
+	OUTGOING_ABILITY_DAMAGE_CAP        GameTag = 284
+	INCOMING_ABILITY_DAMAGE_MULTIPLIER GameTag = 285
+	INCOMING_ABILITY_DAMAGE_CAP        GameTag = 286
+	OUTGOING_COMBAT_DAMAGE_MULTIPLIER  GameTag = 287
+	OUTGOING_COMBAT_DAMAGE_CAP         GameTag = 288
+	INCOMING_COMBAT_DAMAGE_MULTIPLIER  GameTag = 289
+	INCOMING_COMBAT_DAMAGE_CAP         GameTag = 290
+	DIVINE_SHIELD_READY                GameTag = 314
+	IGNORE_DAMAGE_OFF                  GameTag = 354
+	NUM_OPTIONS                        GameTag = 359
+	PENDING_EVOLUTIONS                 GameTag = 461
+
+	// Missing/guessed, only present in logs
+	WEAPON            GameTag = 334
+	DISCARD_CARDS     GameTag = 890
+	CANT_BE_EXHAUSTED GameTag = 244
+	CANT_EXHAUST      GameTag = 226
+	CANT_TARGET       GameTag = 228
+	CANT_DESTROY      GameTag = 229
+
+	CT_INVALID     CardType = 0
+	CT_GAME        CardType = 1
+	CT_PLAYER      CardType = 2
+	CT_HERO        CardType = 3
+	CT_MINION      CardType = 4
+	CT_SPELL       CardType = 5
+	CT_ENCHANTMENT CardType = 6
+	CT_WEAPON      CardType = 7
+	CT_ITEM        CardType = 8
+	CT_TOKEN       CardType = 9
+	CT_HERO_POWER  CardType = 10
+
+	Z_INVALID         Zone = 0
+	Z_PLAY            Zone = 1
+	Z_DECK            Zone = 2
+	Z_HAND            Zone = 3
+	Z_GRAVEYARD       Zone = 4
+	Z_REMOVEDFROMGAME Zone = 5
+	Z_SETASIDE        Zone = 6
+	Z_SECRET          Zone = 7
+
+	BT_ATTACK BlockType = 1
+	BT_JOUST BlockType = 2
+	BT_POWER BlockType = 3
+	BT_TRIGGER BlockType = 5
+	BT_DEATHS BlockType = 6
+	BT_PLAY BlockType = 7
+	BT_FATIGUE BlockType = 8
+	BT_RITUAL BlockType = 9
+	BT_REVEAL_CARD BlockType = 10
+	BT_GAME_RESET BlockType = 11
+
+	// Removed
+	BT_SCRIPT BlockType = 4
+	BT_ACTION BlockType = 99
+
+	// Renamed
+	BT_CONTINUOUS BlockType = 2
+)
+
+var (
+	gameTagNames = []string{}
```

Comment:
Gah, super frustrating that Go doesn't allow these declarations to be constants isn't it?